tests/components/device_tracker/test_unifi_direct.py

```diff
@@ -0,0 +1,169 @@
+from homeassistant.components.device_tracker import (
+    CONF_CONSIDER_HOME, CONF_TRACK_NEW)
+from homeassistant.components.device_tracker.unifi_direct import (
+    DOMAIN, CONF_PORT, PLATFORM_SCHEMA, _response_to_json, get_scanner)
+from homeassistant.const import (CONF_PLATFORM, CONF_PASSWORD, CONF_USERNAME,
+                                 CONF_HOST)
+
+from tests.common import (
+    get_test_home_assistant, assert_setup_component,
+    mock_component, load_fixture)
+
+
+class TestComponentsDeviceTrackerUnifiDirect(unittest.TestCase):
+    """Tests for the Unifi direct device tracker platform."""
+
+    hass = None
+    scanner_path = 'homeassistant.components.device_tracker.' + \
+        'unifi_direct.UnifiDeviceScanner'
+
+    def setup_method(self, _):
+        """Setup things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+        mock_component(self.hass, 'zone')
+
+    def teardown_method(self, _):
+        """Stop everything that was started."""
+        self.hass.stop()
+        try:
+            os.remove(self.hass.config.path(device_tracker.YAML_DEVICES))
+        except FileNotFoundError:
+            pass
+
+    @mock.patch(scanner_path,
+                return_value=mock.MagicMock())
+    def test_get_scanner(self, unifi_mock):  \
+            # pylint: disable=invalid-name
+        """Test creating an Unifi direct scanner with a password."""
+        conf_dict = {
+            DOMAIN: {
+                CONF_PLATFORM: 'unifi_direct',
+                CONF_HOST: 'fake_host',
+                CONF_USERNAME: 'fake_user',
+                CONF_PASSWORD: 'fake_pass',
+                CONF_TRACK_NEW: True,
+                CONF_CONSIDER_HOME: timedelta(seconds=180)
+            }
+        }
+
+        with assert_setup_component(1, DOMAIN):
+            assert setup_component(self.hass, DOMAIN, conf_dict)
+
+        conf_dict[DOMAIN][CONF_PORT] = 22
+        self.assertEqual(unifi_mock.call_args, mock.call(conf_dict[DOMAIN]))
+
+    @patch('pexpect.pxssh.pxssh')
+    def test_get_device_name(self, mock_ssh):
+        """"Testing MAC matching."""
+        conf_dict = {
+            DOMAIN: {
+                CONF_PLATFORM: 'unifi_direct',
+                CONF_HOST: 'fake_host',
+                CONF_USERNAME: 'fake_user',
+                CONF_PASSWORD: 'fake_pass',
+                CONF_PORT: 22,
+                CONF_TRACK_NEW: True,
+                CONF_CONSIDER_HOME: timedelta(seconds=180)
+            }
+        }
+        mock_ssh.return_value.before = load_fixture('unifi_direct.txt')
+        scanner = get_scanner(self.hass, conf_dict)
+        devices = scanner.scan_devices()
+        self.assertEqual(23, len(devices))
+        self.assertEqual("iPhone",
+                         scanner.get_device_name("98:00:c6:56:34:12"))
+        self.assertEqual("iPhone",
+                         scanner.get_device_name("98:00:C6:56:34:12"))
+
+    @patch('pexpect.pxssh.pxssh.login')
```

Comment:
I think you should also patch `logout` so that there's no unnecessary I/O done in the test. Depends on how logout works of course.